linux_os/guide/services/ssh/ssh_server/sshd_use_approved_kex_ordered_stig/oval/shared.xml

```diff
@@ -50,16 +41,38 @@
     </criteria>
   </definition>
 
-  <ind:textfilecontent54_test check="all" check_existence="all_exist"
+  <ind:variable_test check="all" check_existence="all_exist"
     comment="tests the value of Kex algorithms setting in the {{{ path }}} file"
     id="test_{{{ rule_id }}}" version="1">
     <ind:object object_ref="obj_{{{ rule_id }}}" />
-  </ind:textfilecontent54_test>
+    <ind:state state_ref="ste_{{{ rule_id }}}"/>
+  </ind:variable_test>
+
+  <ind:variable_object id="obj_{{{ rule_id }}}" version="1">
+    <ind:var_ref>var_config_{{{ rule_id }}}</ind:var_ref>
+  </ind:variable_object>
+
+  <ind:variable_state comment="approved strong kex" id="ste_{{{ rule_id }}}" version="1">
+    <ind:value operation="equals" datatype="string" var_ref="var_{{{ rule_id }}}" var_check="at least one" />
```

Comment:
I was about to address this. The issue was introduced here: https://github.com/ComplianceAsCode/content/pull/10639

IMO the best option would be just revert the change introduced there in this OVAL file. The issue with this approach is that it doesn't actually ensures the order is correct. There is an issue with the test `correct_scrambled.fail.sh` btw. If you still think this is better and find a way to ensure order is correct, this changes also break tests in OL8 and RHEL8 due to incorrect regexes, I can share which ones if needed.

Reply:
I've reverted that change as suggested as the new oval wouldn't work well with the order.
thanks!